interview-questions.md

```diff
@@ -8,42 +8,46 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: Object-oriented programming focuses on using key/value pairs whereas functional programming puts an emphasis on using higher order functions?
+
+Researched answer: OOP programming is more abstract because it has a focus on operating on keys directly instead of functional programming's declarative approach that requires explicit functions. As such, they have different methods for storing and handling data.
 
-Researched answer:
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: A float is a decimal value and an Integer is a whole number value.
+
+Researched answer: It's a simple question and the research answer confirms that any number written without decimals is an integer and if it has a decimal, it counts as a float. So here's a joke: Why should you never argue with a decimal? It's because they always have a point.
 
-Researched answer:
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: In JavaScript you must explicitly state return in order for scripts to return anything, however, in Ruby, every line has a return, even if it's not displayed to the user.
 
-Researched answer:
+Researched answer: Because Ruby uses implicit returns (returning the value of the last line automatically), people enjoy writting short, elegant code. Explicitly using 'return' can even lead to errors in Ruby!
 
```

Comment:
more content could be discussing those errors or giving an example of implicit return